chainspec.go

```diff
@@ -0,0 +1,141 @@
+package ibctest
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"sync/atomic"
+
+	"github.com/strangelove-ventures/ibctest/ibc"
+)
+
+// ChainSpec is a wrapper around an ibc.ChainConfig
+// that allows callers to easily reference one of the built-in chain configs
+// and optionally provide overrides for some settings.
+type ChainSpec struct {
+	// Name is the name of the built-in config to use as a basis for this chain spec.
+	// Required unless every other field is set.
+	Name string
+
+	// ChainName sets the Name of the embedded ibc.ChainConfig, i.e. the name of the chain.
+	ChainName string
+
+	// Version of the docker image to use.
+	// Must be set.
+	Version string
+
+	// GasAdjustment and NoHostMount are pointers in ChainSpec
+	// so zero-overrides can be detected from omitted overrides.
+	GasAdjustment *float64
+	NoHostMount   *bool
+
+	// Embedded ChainConfig to allow for simple JSON definition of a ChainSpec.
+	ibc.ChainConfig
+
+	// How many validators and how many full nodes to use
+	// when instantiating the chain.
+	// If unspecified, NumValidators defaults to 2 and NumFullNodes defaults to 1.
+	NumValidators, NumFullNodes *int
+
+	// Generate the automatic suffix on demand when needed.
+	autoSuffixOnce sync.Once
+	autoSuffix     string
+}
```

Comment:
Which one of these fields (if any) is guaranteed to be unique per test invocation?